### PR TITLE
Simplify SSE/SSE4.1 intrinsics

### PR DESCRIPTION
**Pull request by fiigii**
We have updated `numArgsOfHWIntrinsics`, so the intrinsics that have variable number of parameters can be table-driven.

### CODE REVIEW

**Comment by fiigii** [comment]
test Windows_NT x64 Checked jitincompletehwintrinsic
test Windows_NT x64 Checked jitx86hwintrinsicnoavx
test Windows_NT x64 Checked jitx86hwintrinsicnoavx2
test Windows_NT x64 Checked jitx86hwintrinsicnosimd
test Windows_NT x64 Checked jitnox86hwintrinsic

test Windows_NT x86 Checked jitincompletehwintrinsic
test Windows_NT x86 Checked jitx86hwintrinsicnoavx
test Windows_NT x86 Checked jitx86hwintrinsicnoavx2
test Windows_NT x86 Checked jitx86hwintrinsicnosimd
test Windows_NT x86 Checked jitnox86hwintrinsic

test Ubuntu x64 Checked jitincompletehwintrinsic
test Ubuntu x64 Checked jitx86hwintrinsicnoavx
test Ubuntu x64 Checked jitx86hwintrinsicnoavx2
test Ubuntu x64 Checked jitx86hwintrinsicnosimd
test Ubuntu x64 Checked jitnox86hwintrinsic

test OSX10.12 x64 Checked jitincompletehwintrinsic
test OSX10.12 x64 Checked jitx86hwintrinsicnoavx
test OSX10.12 x64 Checked jitx86hwintrinsicnoavx2
test OSX10.12 x64 Checked jitx86hwintrinsicnosimd
test OSX10.12 x64 Checked jitnox86hwintrinsic

File: src/jit/lsraxarch.cpp [hunk 1]
```diff
@@ -2342,7 +2342,9 @@ void LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
         case NI_SSE2_CompareNotEqualOrderedScalar:
         case NI_SSE2_CompareNotEqualUnorderedScalar:
             info->internalIntCount = 1;
+#ifdef _TARGET_X86_
             info->setInternalCandidates(this, RBM_BYTE_REGS);
+#endif
```
Comment (fiigii) [inline src/jit/lsraxarch.cpp]: Only 32-bit platforms have this limitation.

**Comment by fiigii** [comment]
@CarolEidt @tannergooding PTAL

File: src/jit/lsraxarch.cpp (hunk 1 above)
Comment (tannergooding) [reply src/jit/lsraxarch.cpp]: Could you please elaborate. The tmpReg is used by `sete`, which requires it to be a byte register.

Comment (mikedn) [reply src/jit/lsraxarch.cpp]: On x64 all registers are byte registers so the call is useless. That said, I think I prefer the existing code, `RBM_BYTE_REGS` is `RBM_ALL_INT` on x64 so the code is correct and one less `ifdef` is a good thing.

Comment (tannergooding) [reply src/jit/lsraxarch.cpp]: >  I think I prefer the existing code

I agree.

Comment (fiigii) [reply src/jit/lsraxarch.cpp]: @mikedn I see, `RBM_BYTE_REGS` already deal with the platform difference. Thank you for pointing out.

**Comment by fiigii** [comment]
test Windows_NT x64 Checked jitincompletehwintrinsic
test Windows_NT x64 Checked jitx86hwintrinsicnoavx
test Windows_NT x64 Checked jitx86hwintrinsicnoavx2
test Windows_NT x64 Checked jitx86hwintrinsicnosimd
test Windows_NT x64 Checked jitnox86hwintrinsic

test Windows_NT x86 Checked jitincompletehwintrinsic
test Windows_NT x86 Checked jitx86hwintrinsicnoavx
test Windows_NT x86 Checked jitx86hwintrinsicnoavx2
test Windows_NT x86 Checked jitx86hwintrinsicnosimd
test Windows_NT x86 Checked jitnox86hwintrinsic

test Ubuntu x64 Checked jitincompletehwintrinsic
test Ubuntu x64 Checked jitx86hwintrinsicnoavx
test Ubuntu x64 Checked jitx86hwintrinsicnoavx2
test Ubuntu x64 Checked jitx86hwintrinsicnosimd
test Ubuntu x64 Checked jitnox86hwintrinsic

File: src/jit/hwintrinsiccodegenxarch.cpp [hunk 2]
```diff
@@ -1070,29 +1052,6 @@ void CodeGen::genSSE41Intrinsic(GenTreeHWIntrinsic* node)
...
-        case NI_SSE41_RoundCurrentDirectionScalar:
-        case NI_SSE41_RoundToNearestIntegerScalar:
-        case NI_SSE41_RoundToNegativeInfinityScalar:
-        case NI_SSE41_RoundToPositiveInfinityScalar:
-        case NI_SSE41_RoundToZeroScalar:
-        {
-            assert((baseType == TYP_FLOAT) || (baseType == TYP_DOUBLE));
-            instruction ins = Compiler::insOfHWIntrinsic(intrinsicID, node->gtSIMDBaseType);
-
-            if (op2 == nullptr)
```
Comment (tannergooding) [inline src/jit/hwintrinsiccodegenxarch.cpp]: Where is the handling for `op2 == nullptr`?

Currently, there is no codepath that will cover `R_R_R_I`

Comment (fiigii) [reply src/jit/hwintrinsiccodegenxarch.cpp]: Now, `numArgsOfHWIntrinsics` automatically return the correct `numArgs`, which 1-arg from will be addressed in its own path.

Comment (fiigii) [reply src/jit/hwintrinsiccodegenxarch.cpp]: >  there is no codepath that will cover R_R_R_I (for the 1 argument case)

This is automatically solved by `IsDstSrcSrcAVXInstruction`.

**Pull request merged**